src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -270,6 +279,100 @@ protected override void Dispose(bool disposing)
+			var isRowVersion = new DataColumn(SchemaTableOptionalColumn.IsRowVersion, typeof(bool));
+			var isUnique = new DataColumn(SchemaTableColumn.IsUnique, typeof(bool));
+			var isKey = new DataColumn(SchemaTableColumn.IsKey, typeof(bool));
+			var isAutoIncrement = new DataColumn(SchemaTableOptionalColumn.IsAutoIncrement, typeof(bool));
+			var isHidden = new DataColumn(SchemaTableOptionalColumn.IsHidden, typeof(bool));
+			var baseCatalogName = new DataColumn(SchemaTableOptionalColumn.BaseCatalogName, typeof(string));
+			var baseSchemaName = new DataColumn(SchemaTableColumn.BaseSchemaName, typeof(string));
+			var baseTableName = new DataColumn(SchemaTableColumn.BaseTableName, typeof(string));
+			var baseColumnName = new DataColumn(SchemaTableColumn.BaseColumnName, typeof(string));
+			var isAliased = new DataColumn(SchemaTableColumn.IsAliased, typeof(bool));
+			var isExpression = new DataColumn(SchemaTableColumn.IsExpression, typeof(bool));
+			var isIdentity = new DataColumn("IsIdentity", typeof(bool));
+			ordinal.DefaultValue = 0;
+			isLong.DefaultValue = false;
+
+			// must maintain order for backward compatibility
+			var columns = schemaTable.Columns;
+			columns.Add(columnName);
+			columns.Add(ordinal);
+			columns.Add(size);
+			columns.Add(precision);
+			columns.Add(scale);
+			columns.Add(isUnique);
+			columns.Add(isKey);
+			columns.Add(baseCatalogName);
+			columns.Add(baseColumnName);
+			columns.Add(baseSchemaName);
+			columns.Add(baseTableName);
+			columns.Add(dataType);
+			columns.Add(allowDBNull);
+			columns.Add(providerType);
+			columns.Add(isAliased);
+			columns.Add(isExpression);
+			columns.Add(isIdentity);
+			columns.Add(isAutoIncrement);
+			columns.Add(isRowVersion);
+			columns.Add(isHidden);
+			columns.Add(isLong);
+			columns.Add(isReadOnly);
+
+			for (var i = 0; i < colDefinitions.Length; ++i)
+			{
+				var col = colDefinitions[i];
+				var schemaRow = schemaTable.NewRow();
+				schemaRow[columnName] = col.Name;
+				schemaRow[ordinal] = i;
+				schemaRow[size] = col.ColumnType.IsString() ? col.ColumnLength / col.CharacterSet.MaxLength() : col.ColumnLength;
+				schemaRow[dataType] = col.GetDataType(Connection.TreatTinyAsBoolean, Connection.OldGuids);
+				schemaRow[providerType] = col.ColumnType;
+				schemaRow[isLong] = col.ColumnLength > 255 && ((col.ColumnFlags & ColumnFlags.Blob) != 0 || col.ColumnType.IsBlob());
+				schemaRow[isKey] = (col.ColumnFlags & ColumnFlags.PrimaryKey) != 0;
+				schemaRow[allowDBNull] = (col.ColumnFlags & ColumnFlags.NotNull) == 0;
+				if (col.ColumnType.IsDecimal())
+				{
+					schemaRow[precision] = col.ColumnLength - 2 + ((col.ColumnFlags & ColumnFlags.Unsigned) != 0 ? 1 : 0);
+					schemaRow[scale] = col.Decimals;
+				}
+
+				schemaRow[baseSchemaName] = col.SchemaName;
+				schemaRow[baseTableName] = col.PhysicalTable;
+				schemaRow[baseColumnName] = col.PhysicalName;
+				schemaRow[isUnique] = false;
```

Comment:
Should this be set from `ColumnFlags.UniqueKey`? It doesn't appear to be covered in the tests to verify that this is the same value as Connector/NET.

Reply:
Oracle set UniqueKey to be `false` always, what's your idea on this backwards compatibility?

Reply:
I'm OK with keeping it hardcoded to `false`; it's easier and backwards-compatible.